src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,110 @@
+#include "sl_lock.h"
+
+#include "sl.h"
+
+void sl_lock_init(struct sl_lock *lock) {
+    assert(lock);
+
+    *lock = (struct sl_lock) {
```

Comment:
This should use your static initializer.

`#define SL_LOCK_STATIC_INIT() (struct sl_lock){ .holder = 0 }`

Casts don't have a space between themselves and what they modify.